android/src/ti/googlesignin/TiGoogleSignInActivity.kt

```diff
@@ -0,0 +1,59 @@
+package ti.googlesignin
+
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.ActivityInfo
+import android.os.Build
+import android.os.Bundle
+import org.appcelerator.kroll.common.Log
+
+class TiGoogleSignInActivity : Activity() {
+
+    private var hasShownSignInActivity = false
+    private var signInIntent: Intent? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        signInIntent = intent.getParcelableExtra("google.sign.in")
+
+        // Allow both landscape- and portrait orientations
+        // TODO: Is this correct?
+        var orientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT or ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
+        if (Build.VERSION.SDK_INT >= 9) {
+            orientation = ActivityInfo.SCREEN_ORIENTATION_SENSOR_PORTRAIT or ActivityInfo.SCREEN_ORIENTATION_SENSOR_LANDSCAPE
+        }
+        requestedOrientation = orientation
```

Comment:
You don't need to set "requestedOrientation" since your activity supports all orientations by default. That is, since your `TiGoogleSignInActivity` in the AndroidManifest.xml does not define an "android:screenOrientation" attribute, it will support all activities. By doing this, your activity will also honor the device's "orientation lock"... although you may want to double check how the orientation lock works with Google Sign-In (it should be fine).

Reply:
Thanks @jquick-axway! Really appreciate it. Do you think it's good to go like this?

Reply:
Yes, removing this part should be fine.

And if you don't want a transition animation when opening an activity, then you can add a `overridePendingTransition(0, 0)` call **after** the `startActivity()`, `startActivityForResult()`, or `activitySupport.launchActivityForResult()` method call.